File: solutions/javascript/difference-of-squares/difference-of-squares.js
export class Squares {
  constructor(number) {
    this.number = number;
  }

  get sumOfSquares() {
    let sum = 0;
    for (let i = 1; i <= this.number; i++)
      sum += i ** 2;
    return sum;
  }

  get squareOfSum() {
    let sum = 0;
    for (let i = 1; i <= this.number; i++)
      sum += i;
    return sum ** 2;
  }

  get difference() {
    return this.squareOfSum - this.sumOfSquares;
  }
}

File: solutions/ruby/difference-of-squares/difference_of_squares.rb
class Squares
  private

  attr_accessor :number

  def initialize(number)
    self.number = number
  end

  public

  def square_of_sum()= (1..number).sum**2

  def sum_of_squares()= (1..number).sum { |n| n * n }

  def difference()= square_of_sum - sum_of_squares
end
